src/main/haskell/language-kore/test/parser/Data/Kore/ASTUtils/ASTUtilsTest.hs

```diff
@@ -0,0 +1,171 @@
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE Rank2Types             #-}
+{-# LANGUAGE AllowAmbiguousTypes    #-}
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE BangPatterns           #-}
+{-# LANGUAGE LambdaCase             #-}
+{-# LANGUAGE DeriveFunctor          #-}
+{-# LANGUAGE DeriveFoldable         #-}
+{-# LANGUAGE DeriveTraversable      #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE PatternSynonyms        #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeApplications       #-}
+{-# LANGUAGE StandaloneDeriving     #-}
+{-# LANGUAGE TypeSynonymInstances   #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+{-# OPTIONS_GHC -Wno-missing-pattern-synonym-signatures #-}
+
+module  Data.Kore.ASTUtils.ASTUtilsTest
+    ( astUtilsTests
+    ) where
+
+import           Test.Tasty                                          (TestTree,
+                                                                      testGroup)
+
+import           Test.Tasty.HUnit                 (assertEqual, testCase)
+
+
+import           Control.Lens
+import           Data.Reflection
+
+import           Data.Kore.IndexedModule.MetadataTools
+import           Data.Kore.AST.Common 
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.AST.PureML
+import           Data.Kore.ASTUtils.SmartConstructors
+import           Data.Kore.ASTUtils.Substitution
+
+astUtilsTests :: TestTree
+astUtilsTests = testGroup "astUtils" [ substitutions, sortAgreement ]
+
+substitutions :: TestTree
+substitutions = testGroup "substitutions" 
+  [ testCase "subTrivial" $ assertEqual "" subTrivial subTrivialSolution
```

Comment:
This project uses 4 spaces for indentation almost everywhere, "where" and "=" (only sometimes and only when used in function definitions) are the only exceptions I can think of right now. The main reasons for using 4 spaces are uniformity and the fact that "where" and "=" use half of a normal indentation and it looks bad when they are indented by only one space. Note that function signatures are also indented by 4 spaces instead of 2.

BTW, I think that there are places above where you're using a one-space indentation.